explain-indexes.md

```diff
@@ -0,0 +1,343 @@
+| │ └─IndexFullScan_18           | 10.00   | cop[tikv] | table:t1, index:intkey(intkey) | keep order:true, desc              |
+| └─TableRowIDScan_19(Probe)     | 10.00   | cop[tikv] | table:t1                       | keep order:false, stats:pseudo     |
++--------------------------------+---------+-----------+--------------------------------+------------------------------------+
+4 rows in set (0.00 sec)
+
+```
+
+In the above example, the last 20 rows are read from the index `intkey`. These `RowID` values are then retrieved from the table data.
+
+## IndexReader
+
+TiDB supports the _covering index optimization_. If all rows can be retrieved from an index, TiDB will skip the second step that is usually required in an `IndexLookup`. Consider the following two examples:
+
+{{< copyable "sql" >}}
+
+```sql
+EXPLAIN SELECT * FROM t1 WHERE intkey = 123;
+EXPLAIN SELECT id FROM t1 WHERE intkey = 123;
+```
+
+```sql
++-------------------------------+---------+-----------+--------------------------------+-----------------------------------+
+| id                            | estRows | task      | access object                  | operator info                     |
++-------------------------------+---------+-----------+--------------------------------+-----------------------------------+
+| IndexLookUp_10                | 1.00    | root      |                                |                                   |
+| ├─IndexRangeScan_8(Build)     | 1.00    | cop[tikv] | table:t1, index:intkey(intkey) | range:[123,123], keep order:false |
+| └─TableRowIDScan_9(Probe)     | 1.00    | cop[tikv] | table:t1                       | keep order:false                  |
++-------------------------------+---------+-----------+--------------------------------+-----------------------------------+
+3 rows in set (0.00 sec)
+
++--------------------------+---------+-----------+--------------------------------+-----------------------------------+
+| id                       | estRows | task      | access object                  | operator info                     |
++--------------------------+---------+-----------+--------------------------------+-----------------------------------+
+| Projection_4             | 1.00    | root      |                                | test.t1.id                        |
+| └─IndexReader_6          | 1.00    | root      |                                | index:IndexRangeScan_5            |
+|   └─IndexRangeScan_5     | 1.00    | cop[tikv] | table:t1, index:intkey(intkey) | range:[123,123], keep order:false |
++--------------------------+---------+-----------+--------------------------------+-----------------------------------+
+3 rows in set (0.00 sec)
+```
+
+Because `id` is also the internal `RowID`, it is stored in the `intkey` index. After using the `intkey` index as part of `└─IndexRangeScan_5`, the value of the `RowID` can be returned directly.
```

Comment:
```suggestion
Because `id` is also the internal `RowID`, `id` is stored in the `intkey` index. After using the `intkey` index as a part of the `└─IndexRangeScan_5` operator task, the value of the `RowID` can be returned directly.
```